### PR TITLE
Add support for custom network on-connect messages

### PR DESCRIPTION
**Pull request by delthas**
Some servers use custom IRC bots with custom commands for registering to
specific services after connection.

This adds support for setting a custom raw IRC message, that will be
sent after registering to a network.

### CODE REVIEW

**Comment by delthas** [comment]
@emersion : thoughts? This feature is important to me but I'm not sure how to implement it (a flag in `network create`? a custom service command? should we support a list of messages or just one? wording (is `-connect-message` ok?))

Also of note that since we don't have `network modify` yet, setting the connect message would require deleting and creating the network again, but after `network modify` is added this should be a bit less tedious.

**Comment by delthas** [comment]
By the way, excellent job on the db migration design, I can't belive adding a column is as easy as adding an `alter table` statement in the migrations array. 👍

**Comment by emersion** [comment]
I'm not against the idea, sadly some networks still don't support SASL. Note that SASL is superior because it provides better error handling and authenticates before registration completes (NickServ causes issues when the user wants to join restricted channels for instance, like #freebsd on Freenode).

Because this is more-or-less legacy support, I'd prefer not to do anything too complicated to support this feature.

Are there other potential use-cases than authentication?

Maybe a better name would be `post_register_command`? (Commands sent after registration is complete) But maybe that would just be more confusing from a user PoV, not sure.

>This feature is important to me but I'm not sure how to implement it (a flag in network create? a custom service command?

A network flag sounds best.

>should we support a list of messages or just one? 

Maybe store messages separated by `\r\n` in the DB, and allow the user to configure them via multiple flags? (e.g. `-connect-command "PRIVMSG nick1 hey there" -connect-command "PRIVMSG nick2 hey there"`) But then how to unset the list? Or maybe require a literal `\n` (`-connect-command "PRIVMSG nick1 hey there\nRIVMSG nick2 hey there"`)

**Comment by delthas** [comment]
> Are there other potential use-cases than authentication?
Yes-ish: some servers or chanels use a custom IRC bot, distinct from NickServ, that you send a privmsg to in order to join a channel.

> Maybe store messages separated by \r\n in the DB, and allow the user to configure them via multiple flags?
👍 

> But then how to unset the list?
`-connect-command ""` (same as other fields)

**Comment by emersion** [comment]
This plan looks good to me.

**Comment by delthas** [comment]
Updated. Note that I've put the string split/join in the db methods to "abstract away" its storage as a `\r\n` string. The other solution is to split it each time when sending it after registering. I think it doesn't really matter performance-wise.

File: service.go [hunk 1]
```diff
@@ -174,6 +174,17 @@ func newFlagSet() *flag.FlagSet {
 	return fs
 }
 
+type StringSliceVar []string
```
Comment (emersion) [inline service.go:177]: Let's keep this type unexported

**Pull request merged**